src/ansys/geometry/core/geometry/surfaces/plane.py

```diff
@@ -0,0 +1,230 @@
+# SOFTWARE.
+"""Provides for creating and managing a cylinder."""
+from functools import cached_property
+
+from beartype import beartype as check_input_types
+from beartype.typing import Union
+import numpy as np
+
+from ansys.geometry.core.geometry.parameterization import (
+    Interval,
+    Parameterization,
+    ParamForm,
+    ParamType,
+    ParamUV,
+)
+from ansys.geometry.core.geometry.surfaces.surface import Surface
+from ansys.geometry.core.geometry.surfaces.surface_evaluation import SurfaceEvaluation
+from ansys.geometry.core.math import (
+    UNITVECTOR3D_X,
+    UNITVECTOR3D_Z,
+    Matrix44,
+    Point3D,
+    UnitVector3D,
+    Vector3D,
+)
+from ansys.geometry.core.typing import Real, RealSequence
+
+
+class PlaneSurface(Surface):
+    """
+    Provides 3D plane surface representation.
+
+    Parameters
+    ----------
+    origin : Union[~numpy.ndarray, RealSequence, Point3D],
+        Centered origin of the torus.
+    reference : Union[~numpy.ndarray, RealSequence, UnitVector3D, Vector3D]
+        X-axis direction.
+    axis : Union[~numpy.ndarray, RealSequence, UnitVector3D, Vector3D]
+        X-axis direction.
+    """
+
+    def __init__(
+        self,
+        origin: Union[np.ndarray, RealSequence, Point3D],
+        reference: Union[np.ndarray, RealSequence, UnitVector3D, Vector3D] = UNITVECTOR3D_X,
+        axis: Union[np.ndarray, RealSequence, UnitVector3D, Vector3D] = UNITVECTOR3D_Z,
+    ):
+        """Initialize ``PlaneSurface`` class."""
+        self._origin = Point3D(origin) if not isinstance(origin, Point3D) else origin
+
+        self._reference = (
+            UnitVector3D(reference) if not isinstance(reference, UnitVector3D) else reference
+        )
+        self._axis = UnitVector3D(axis) if not isinstance(axis, UnitVector3D) else axis
+        if not self._reference.is_perpendicular_to(self._axis):
+            raise ValueError("Plane reference (dir_x) and axis (dir_z) must be perpendicular.")
+
+    @property
+    def origin(self) -> Point3D:
+        """Origin of the cylinder."""
+        return self._origin
+
+    @property
+    def dir_x(self) -> UnitVector3D:
+        """X-direction of the cylinder."""
+        return self._reference
+
+    @property
+    def dir_y(self) -> UnitVector3D:
+        """Y-direction of the cylinder."""
+        return self.dir_z.cross(self.dir_x)
+
+    @property
+    def dir_z(self) -> UnitVector3D:
+        """Z-direction of the cylinder."""
+        return self._axis
+
+    @check_input_types
+    def __eq__(self, other: "PlaneSurface") -> bool:
+        """Check whether two planes are equal."""
+        return (
+            self._origin == other._origin
+            and self._reference == other._reference
+            and self._axis == other._axis
+        )
+
+    def contains_param(self, param_uv: ParamUV) -> bool:
+        """Check whether the plane contains a u and v pair point."""
```

Comment:
See here "u and v" are used--not capitalized and not tagged as a code entity. We should really be consistent.